### PR TITLE
provider/common tests

### PR DESCRIPTION
**Pull request by BenTheElder**
this is PR #919 + cleanup

### CODE REVIEW

**Comment by BenTheElder** [comment]
rebased, and increased coverage to 100%, also fixed some behavior slightly while inspecting odd existing test case values in GetProxyEnvs

**Comment by aojea** [comment]
/lgtm

thanks for the fixes  😅

File: pkg/internal/cluster/providers/provider/common/proxy.go [hunk 1]
```diff
@@ -48,7 +52,11 @@ func GetProxyEnvs(cfg *config.Cluster) map[string]string {
 	}
 	// Specifically add the cluster subnets to NO_PROXY if we are using a proxy
 	if len(envs) > 0 {
-		noProxy := strings.Join([]string{envs[NOProxy], cfg.Networking.ServiceSubnet, cfg.Networking.PodSubnet}, ",")
```
Comment (BenTheElder) [inline pkg/internal/cluster/providers/provider/common/proxy.go]: previously we would do something like `,10.0.0.0/24,12.0.0.0/24` if `NO_PROXY` was not found which technically seems to have worked (?) but at best seems strange and at worse seems likely to break some application, given how under-defined this environment variable "API" is.

Comment (aojea) [reply pkg/internal/cluster/providers/provider/common/proxy.go]: I remember vaguely to have verified it ... but don't remember how 😅

**Pull request merged**